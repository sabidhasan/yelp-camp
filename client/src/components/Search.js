import React from 'react'
import { Route } from 'react-router-dom'
import PropTypes from 'prop-types';
import { parse } from 'query-string'
import haversine from 'haversine'

import { provinces, activitySymbols, shortenDescription } from '../helpers/helpers'
import SearchResultTile from './SearchResultTile'
import SearchBar from './SearchBar'
import FilterSearch from './FilterSearch'
import Pagination from './Pagination'
import SearchResultsMap from './SearchResultsMap'

class Search extends React.Component {
  constructor(props) {
    super(props)
    // Static variables (well, originalResults is not static!)
    this.searchQuery = parse(this.props.location.search)['q'];
    this.originalResults = [];

    this.state = {
      filteredResults: [],
      page: Math.max(0, parseInt(parse(this.props.location.search)['start'])) || 0,
      hovered: null,
      filterCriteria: null
    };

    // Function bindings
    this.updateFilteredResults = this.updateFilteredResults.bind(this)
    this.goToPage = this.goToPage.bind(this)
  }

  // Holds location of the user if available, register and deregister withLoading
  static contextTypes = {
    userLocation: PropTypes.object,
    startLoad: PropTypes.func,
    finishLoad: PropTypes.func
  };

  componentDidMount() {
    this.context.startLoad(this.constructor.name);
    fetch(`/search?q=${this.searchQuery}`)
      .then(res => res.json())
      .then(search => {
        search = search.map(result => {
          const cgLoc = {latitude: result.lat, longitude: result.lon}
          if (this.context.userLocation && this.context.userLocation.latitude && this.context.userLocation.longitude && result.lat && result.lon) {
            result.distanceFromUser = Math.round(haversine(cgLoc, this.context.userLocation))
          }
          return result;
        });
        // Update original results variable and generate select box variables from that
        this.originalResults = search;
        // this.makeFilterVariables();

        // Update maxDistance, activities and regions, provinces
        const _fc = {regions: [], activities: [], provinces: [], maxDistance: 0};
        _fc.provinces = Object.keys(provinces).map(v => ({value: v, text: provinces[v]}));
        search.forEach(r => {
          _fc.maxDistance = Math.max(_fc.maxDistance, r.distanceFromUser || 0);
          r.activities.forEach(v => {if (!_fc.activities.includes(v) && v) _fc.activities.push(v)});
          if (!_fc.regions.includes(r.region) && r.region) _fc.regions.push(r.region)
        });

        this.setState({
          filterCriteria: _fc,
          filteredResults: search,
          page: Math.min(Math.ceil(search.length / 10) - 1, this.state.page)}, () => this.context.finishLoad(this.constructor.name)
        );
      })
  }

  goToPage(e, page) {
    e.preventDefault();
    this.setState({ page })
  }

  updateFilteredResults(val) {
    // console.log(val);
    // Val contains info on whats selected, you apply the filter here.
    const filteredResults = [];
    for (let item in this.originalResults) {
      // Filter based on province, region, activities
      const _cI = this.originalResults[item];
      if (
        (!val.selectedProv || provinces[val.selectedProv].toLowerCase() === _cI.province.toLowerCase()) &&
        (val.selectedRegions.includes(_cI.region) || (!_cI.region && this.state.filterCriteria.regions.length === val.selectedRegions.length)) &&
        (_cI.activities.some(v => val.selectedActivities.includes(v)) || (!_cI.activities.length && this.state.filterCriteria.activities.length === val.selectedActivities.length)) &&
        (!_cI.distanceFromUser || (_cI.distanceFromUser >= val.selectedDistances.min && _cI.distanceFromUser <= val.selectedDistances.max))
        ) {
          filteredResults.push(_cI)
        }
    }
    // Sort if needed
    if (val.sortBy === 'Distance') {
      filteredResults.sort((a, b) => {
        // Force the missing distances to the bottom, and sort others
        if (!a.distanceFromUser) return 1;
        if (!b.distanceFromUser) return -1;
        return a.distanceFromUser > b.distanceFromUser ? 1 : -1;
      });
    }
    this.setState({filteredResults: filteredResults, page: 0})
  }

  render() {
    const results = this.state.filteredResults
      .slice(this.state.page * 10, (this.state.page * 10) + 10)
      .map((r, idx) => {
        return (
          <SearchResultTile
            key={r.id}
            number={(this.state.page * 10) + idx + 1}
            name={r.campgroundName}
            regionAndProvince={[r.region, r.province].filter(a=>!!a).join(', ')}
            address={shortenDescription(r.address)}
            description={shortenDescription(r.description)}
            rating={(r.comments && r.comments.reduce((acc, val) => acc + val.rating, 0) / r.comments.length) || 0}
            comments={r.comments}
            images={r.images}
            id={r.id}
            distance={r.distanceFromUser}
            onTitleHover={(id) => this.setState({hovered: id})}
          />
        )
      })

    const currentResultEndUserFriendly = Math.min((this.state.page * 10) + 10, this.state.filteredResults.length)
    return (
      <div className='Search'>
        <div className='Search__filter-container'>
          {!this.originalResults.length && this.searchQuery ?
            <h1>No results found for {this.searchQuery}</h1>
            : !this.originalResults.length && !this.searchQuery ?
              <h1>Search</h1>
              : <h1>{`Search for ${this.searchQuery}`}</h1>
          }
          <Route
            path="/"
            render={(props) => <SearchBar {...props} />}
          />
          {this.state.filterCriteria ?
            <FilterSearch
              filterCriteria={this.state.filterCriteria}
              className={`${!this.originalResults.length ? ' FilterSearch--hidden' : ''}`}
              onChange={val => this.updateFilteredResults(val)}
            />
          : null
          }
        </div>
        <Pagination
          currentPage={this.state.page + 1}  goToPageHandler={this.goToPage}
          lastPage={Math.ceil(this.state.filteredResults.length / 10)}
          totalResults={this.state.filteredResults.length}
        />
        <div className='Search__results'>
          <h2 className='Search__summary'>
            {currentResultEndUserFriendly > 0
              ? <React.Fragment>
                  Showing results {(this.state.page * 10) + 1}-{currentResultEndUserFriendly}
                  {' '} of {this.state.filteredResults.length}
                  {' '} for <span className='bold'>{this.searchQuery}</span>
                </React.Fragment>
              : this.searchQuery ?
                <React.Fragment>No campgrounds found with the applied filter(s)</React.Fragment>
                : null
              }
          </h2>
          <ol className='Search__results-list'>
            {results}
          </ol>
          <SearchResultsMap
            results={this.state.filteredResults}
            page={this.state.page}
            hovered={this.state.hovered} />
        </div>
      </div>
      )
  }
}

export default Search
