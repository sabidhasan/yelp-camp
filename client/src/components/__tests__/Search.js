import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Search from '../Search'

configure({ adapter: new Adapter() });

jest.mock('../../services/fetchSearch')

describe('<Search />', () => {
  let component;
  let props;
  const context = {
    startLoad: jest.fn(),
    finishLoad: jest.fn(),
    userLocation: undefined
  }
  const onChange = jest.fn();

  beforeEach(() => {
    props = {
      location: {
        search: '?q=Test Search'
      }
    }
    component = shallow(<Search { ...props } />, {
      context
    })
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('When there are 25 search results, default page (0) shows 10', (done) => {
    expect(component.find('SearchResultTile').length).toBe(10);
    done();
  })

  it('When there are 25 search results, changing to valid page number (2) works', (done) => {
    component.instance().goToPage({preventDefault: jest.fn()}, 2)
    expect(component.state('page')).toBe(2);
    done();
  })

  it('When there are 25 search results, changing invalid page number fails', (done) => {
    component.instance().goToPage({preventDefault: jest.fn()}, 3)
    expect(component.state('page')).toBe(0);
    done();
  })

  it('Giving valid param page number shows results', (done) => {
    let props = {
      location: {
        search: '?q=Test Search&start=2'
      }
    }
    component = shallow(<Search { ...props } />, {
      context: {
        startLoad: jest.fn(),
        finishLoad: jest.fn(),
        userLocation: undefined
      }
    })
    // setTimeout is needed for async function
    setTimeout(() => {
      try {
        expect(component.state('page')).toBe(2);
        done();
      } catch (e) {
        done.fail(e)
      }
    })
  })

  it('Giving too high of a page number defaults to last page', (done) => {
    let props = {
      location: {
        search: '?q=Test Search&start=20'
      }
    }
    component = shallow(<Search { ...props } />, {
      context: {
        startLoad: jest.fn(),
        finishLoad: jest.fn(),
        userLocation: undefined
      }
    })
    setTimeout(() => {
      try {
        expect(component.state('page')).toBe(2);
        done();
      } catch (e) {
        done.fail(e)
      }
    })
  })

  it('When there are 25 search results, giving invalid param page number', (done) => {
    expect(component.find('SearchResultTile').length).toBe(10);
    done();
  })

  it('Page 2 result #1 is numbered 11', (done) => {
    component.instance().goToPage({preventDefault: jest.fn()}, 1)
    try {
      expect(component.find('SearchResultTile').first().props().number).toBe(11)
      done()
    } catch (err) {
      done.fail(err)
    }
  })

  it('Page 2 result #10 is numbered 20', (done) => {
    component.instance().goToPage({preventDefault: jest.fn()}, 1)
    try {
      expect(component.find('SearchResultTile').at(9).props().number).toBe(20)
      done()
    } catch (err) {
      done.fail(err)
    }
  })

  it('Filters are shown', () => {
    expect(component.find('FilterSearch').length).toBe(1)
  })

  it('Applying filter performs filtration', () => {
    const o = {
      selectedProv: null,
      selectedActivities: ['laundromat'],
      selectedRegions: ['Thunder Bay'],
      selectedDistances: { min: 0, max: 40 },
      sortBy: 'Distance',
    }
    component.instance().updateFilteredResults(o)
    expect(component.find('SearchResultTile').length).toBe(5);
  })

  it('MultiCheckBox component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
})
