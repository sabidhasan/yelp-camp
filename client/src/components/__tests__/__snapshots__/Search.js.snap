// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> MultiCheckBox component snapshot matches what is expected 1`] = `
<div
  className="Search"
>
  <div
    className="Search__filter-container"
  >
    <h1>
      Search for Test Search
    </h1>
    <FilterSearch
      className=""
      filterCriteria={
        Object {
          "activities": Array [
            "cable tv hookups",
            "pull thru sites",
            "full hookup sites",
            "tenting sites",
            "dumping station",
            "open all year",
            "flush toilets",
            "pets allowed",
            "laundromat",
          ],
          "maxDistance": 0,
          "provinces": Array [
            Object {
              "text": "british columbia",
              "value": "bc",
            },
            Object {
              "text": "alberta",
              "value": "ab",
            },
            Object {
              "text": "saskatchewan",
              "value": "sk",
            },
            Object {
              "text": "manitoba",
              "value": "mb",
            },
            Object {
              "text": "ontario",
              "value": "on",
            },
            Object {
              "text": "quebec",
              "value": "qc",
            },
            Object {
              "text": "prince edward island",
              "value": "pe",
            },
            Object {
              "text": "new brunswick",
              "value": "nb",
            },
            Object {
              "text": "nova scotia",
              "value": "ns",
            },
            Object {
              "text": "newfoundland",
              "value": "nl",
            },
            Object {
              "text": "yukon territory",
              "value": "yt",
            },
            Object {
              "text": "northwest territories",
              "value": "nt",
            },
            Object {
              "text": "nunavut",
              "value": "nu",
            },
          ],
          "regions": Array [
            "Montreal",
            "North Lake",
            "Kelowna",
            "Thunder Bay",
            "Edmonton",
          ],
        }
      }
      onChange={[Function]}
    />
  </div>
  <Pagination
    currentPage={1}
    goToPageHandler={[Function]}
    lastPage={3}
    totalResults={25}
  />
  <div
    className="Search__results"
  >
    <h2
      className="Search__summary"
    >
      Showing results 1-10 of
                       25 for
      <span
        className="bold"
      >
        Test Search
      </span>
    </h2>
    <ol
      className="Search__results-list"
    >
      <SearchResultTile
        address=""
        description=""
        id={50}
        key="50"
        name="Glowing Embers RV Park & Travel Centre"
        number={1}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="Montreal, Quebec"
      />
      <SearchResultTile
        address=""
        description=""
        id={51}
        key="51"
        name="Glowing Embers RV Park & Travel Centre"
        number={2}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="North Lake, Manitoba"
      />
      <SearchResultTile
        address=""
        description=""
        id={52}
        key="52"
        name="Glowing Embers RV Park & Travel Centre"
        number={3}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="Kelowna, British Columbia"
      />
      <SearchResultTile
        address=""
        description=""
        id={53}
        key="53"
        name="Glowing Embers RV Park & Travel Centre"
        number={4}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="Thunder Bay, Ontario"
      />
      <SearchResultTile
        address=""
        description=""
        id={54}
        key="54"
        name="Glowing Embers RV Park & Travel Centre"
        number={5}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="Edmonton, Alberta"
      />
      <SearchResultTile
        address=""
        description=""
        id={55}
        key="55"
        name="Glowing Embers RV Park & Travel Centre"
        number={6}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="Kelowna, British Columbia"
      />
      <SearchResultTile
        address=""
        description=""
        id={56}
        key="56"
        name="Glowing Embers RV Park & Travel Centre"
        number={7}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="Montreal, Quebec"
      />
      <SearchResultTile
        address=""
        description=""
        id={57}
        key="57"
        name="Glowing Embers RV Park & Travel Centre"
        number={8}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="Thunder Bay, Ontario"
      />
      <SearchResultTile
        address=""
        description=""
        id={58}
        key="58"
        name="Glowing Embers RV Park & Travel Centre"
        number={9}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="Kelowna, British Columbia"
      />
      <SearchResultTile
        address=""
        description=""
        id={59}
        key="59"
        name="Glowing Embers RV Park & Travel Centre"
        number={10}
        onTitleHover={[Function]}
        rating={0}
        regionAndProvince="Edmonton, Alberta"
      />
    </ol>
    <Wrapper
      hovered={null}
      page={0}
      results={
        Array [
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 50,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Quebec",
            "region": "Montreal",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 51,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Manitoba",
            "region": "North Lake",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 52,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "British Columbia",
            "region": "Kelowna",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 53,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Ontario",
            "region": "Thunder Bay",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 54,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Alberta",
            "region": "Edmonton",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 55,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "British Columbia",
            "region": "Kelowna",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 56,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Quebec",
            "region": "Montreal",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 57,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Ontario",
            "region": "Thunder Bay",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 58,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "British Columbia",
            "region": "Kelowna",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 59,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Alberta",
            "region": "Edmonton",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 60,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Quebec",
            "region": "Montreal",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 61,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Ontario",
            "region": "Thunder Bay",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 62,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Quebec",
            "region": "Montreal",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 63,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Manitoba",
            "region": "North Lake",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 64,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Alberta",
            "region": "Edmonton",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 65,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Ontario",
            "region": "Thunder Bay",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 66,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Quebec",
            "region": "Montreal",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 67,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "British Columbia",
            "region": "Kelowna",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 68,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Quebec",
            "region": "Montreal",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 69,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Alberta",
            "region": "Edmonton",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 70,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "British Columbia",
            "region": "Kelowna",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 71,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Manitoba",
            "region": "North Lake",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 72,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Quebec",
            "region": "Montreal",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 73,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Ontario",
            "region": "Thunder Bay",
          },
          Object {
            "activities": Array [
              "cable tv hookups",
              "pull thru sites",
              "full hookup sites",
              "tenting sites",
              "dumping station",
              "open all year",
              "flush toilets",
              "pets allowed",
              "laundromat",
            ],
            "campgroundName": "Glowing Embers RV Park & Travel Centre",
            "id": 74,
            "lat": 53.541219,
            "lon": -113.775276,
            "province": "Alberta",
            "region": "Edmonton",
          },
        ]
      }
    />
  </div>
</div>
`;
